mlflow/models/evaluation/base.py

```diff
@@ -598,6 +661,107 @@ def _get_last_failed_evaluator():
+    }
+
+    for metric_name in validation_thresholds.keys():
+
+        metric_threshold, validation_result = (
+            validation_thresholds[metric_name],
+            validation_results[metric_name],
+        )
+
+        if metric_name not in candidate_metrics:
+            validation_result.missing_candidate = True
+            continue
+
+        candidate_metric_value, baseline_metric_value = (
+            candidate_metrics[metric_name],
+            baseline_metrics[metric_name] if baseline_metrics else None,
+        )
+
+        # If metric is higher is better, >= is used, otherwise <= is used
+        # for thresholding metric value and model comparsion
+        comparator_fn = operator.__ge__ if metric_threshold.higher_is_better else operator.__le__
+
+        if metric_threshold.threshold is not None:
+            # metric threshold failed
+            # - if not (metric_value >= threshold) for higher is better
+            # - if not (metric_value <= threshold) for lower is better
+            validation_result.threshold_failed = not comparator_fn(
+                candidate_metric_value, metric_threshold.threshold
+            )
+
+        if (
+            metric_threshold.min_relative_change or metric_threshold.min_absolute_change
+        ) and metric_name not in baseline_metrics:
+            validation_result.missing_baseline = True
+            continue
+
+        if metric_threshold.min_absolute_change is not None:
+            # metric comparsion aboslute change failed
+            # - if not (metric_value >= baseline + min_absolute_change) for higher is better
+            # - if not (metric_value <= baseline - min_absolute_change) for lower is better
+            if metric_threshold.higher_is_better:
+                validation_result.min_absolute_change_failed = not comparator_fn(
+                    Decimal(candidate_metric_value),
+                    Decimal(baseline_metric_value + metric_threshold.min_absolute_change),
+                )
+            else:
+                validation_result.min_absolute_change_failed = not comparator_fn(
+                    Decimal(candidate_metric_value),
+                    Decimal(baseline_metric_value - metric_threshold.min_absolute_change),
+                )
+
+        if metric_threshold.min_relative_change is not None:
+            # metric comparsion relative change failed
+            # - if (metric_value - baseline) / baseline < min_relative_change for higher is better
+            # - if (baseline - metric_value) / baseline < min_relative_change for lower is better
+            if metric_threshold.higher_is_better:
+                relative_change = (
+                    candidate_metric_value - baseline_metric_value
+                ) / baseline_metric_value
+            else:
+                relative_change = (
+                    baseline_metric_value - candidate_metric_value
+                ) / baseline_metric_value
```

Comment:
These computations fail when `baseline_metric_value` is zero, which is the case in the following overfitting example:

```
import mlflow
from mlflow.models.evaluation import EvaluationDataset
from mlflow.models import MetricThreshold

from sklearn import datasets
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

# prepare train and test data
iris = datasets.load_iris()
X = iris.data
y = iris.target
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

import pandas as pd
xy_test = pd.DataFrame(X_test, columns=["0", "1", "2", "3"])
xy_test["label"] = y_test

thresholds = {
    # Metric Value Threshold
    "root_mean_squared_error": MetricThreshold(
        threshold=0.05,
        higher_is_better=False,
        min_relative_change=0.1
    ),
}

with mlflow.start_run():
    rid = mlflow.active_run().info.run_id

    logistic = LogisticRegression(C=1.0)
    logistic.fit(X_train, y_train)
    mod = mlflow.sklearn.log_model(logistic, artifact_path="mymodel")
    print(mod.model_uri)

    results = mlflow.evaluate(
        model="runs:/%s/mymodel" % rid,
        model_type="regressor",
        data=xy_test,
        evaluator_config={
            "log_model_explainability": False,
        },
        evaluators="default",
        validation_thresholds=thresholds,
        targets="label",
        baseline_model="runs:/acbea54588ee4122add50991d7094efd/mymodel",
    )
    print("RES", type(results))
    print(results.metrics)
    print(results.baseline_model_metrics)
    print(results.artifacts)
```

Reply:
Traceback (most recent call last):
  File "eval.py", line 36, in <module>
    results = mlflow.evaluate(
  File "/Users/corey.zumar/mlflow/mlflow/models/evaluation/base.py", line 1189, in evaluate
    _validate(
  File "/Users/corey.zumar/mlflow/mlflow/models/evaluation/base.py", line 747, in _validate
    relative_change = (
ZeroDivisionError: float division by zero

Reply:
I've adapted your feedback. Here's the output of the overfitting example.
<img width="1008" alt="Screen Shot 2022-08-23 at 12 17 39 PM" src="https://user-images.githubusercontent.com/66143562/186254831-e5be753b-2d4f-4992-89be-6311b44ba245.png">